File: fixtures/lang-coverage/kotlin/src/Service.kt
// feature: class, imports (single, alias, star), cross-file call, external call
package com.example

import java.util.LinkedHashMap
import java.util.Collections as Col
import kotlin.math.*

class Service(private val name: String) : Greeter {
    private val cache = LinkedHashMap<String, String>()

    override fun hi(): String {
        return "hello from " + name
    }

    fun run() {
        val greeting = formatName(name)
        println(greeting)
        Col.emptyMap<String, String>()
    }
}

File: fixtures/lang-coverage/go/service.go
// feature: struct method, grouped import, aliased import, cross-file call,
//          external call (fmt.Println must not produce local edge).
package graphy

import (
	"fmt"
	f "fmt"
	_ "os"
)

// Service satisfies Greeter structurally.
type Service struct {
	name string
}

// NewService constructs a Service.
func NewService(name string) *Service {
	return &Service{name: name}
}

// Hi implements the Greeter interface.
func (s *Service) Hi() string {
	return "hello from " + s.name
}

// Run executes the service logic.
func (s *Service) Run() {
	greeting := FormatName(s.name)
	fmt.Println(greeting)
	f.Println("done")
}
